### PR TITLE
Add Elixir extension

### PR DESCRIPTION
**Pull request by dinocosta**
## Description

Introduce a new extension that allows searching and reading Elixir's standard library documentation without leaving Raycast.

This extension introduces has two different commands:

* `Search Functions` – Lists all functions, types, macros and callbacks available in Elixir's standard library, with a different section for each module. Users can type the module and function to find a specific function, as well as filter by type (Function, Type, Macro, Callback).
* `Search Modules` – Lists all modules for which documentation is available, allowing the users to navigate to a module's list of functions, types, macros and callbacks, where users can, once again, filter by name and type.

> [!NOTE]
> Although there's a lot of files added, most of them are simple Javascript objects with the module's documentation. An Elixir script is also included that automatically generates this documentation files – `doc_builder.exs` – but I feel like this should probably not be a part of the extension itself, but maybe something that is kept in a Github Gist or a different repository altogether.

## Screencast

These screenshots are available in the `metadata` folder but I'll also leave them here with relevant captions 🔽 

<details>
<summary>Search Functions</summary>

elixir-1
*The "Search Functions" command lists all functions, types, macros and callbacks in Elixir's standard library with a search field as well as a type dropdown*
</details>

<details>
<summary>Search Functions: Type Dropdown</summary>

elixir-2
*Users can filter by type in the "Search Functions" command*
</details>

<details>
<summary>Search Functions: Text Search</summary>

elixir-3
*Users can search by name in the "Search Functions" command*
</details>

<details>
<summary>Function Detail</summary>

elixir-4
*The function's documentation is displayed using Markdown, so we can display code blocks and more closely resemble what users can see in the official documentation*
</details>

<details>
<summary>Function Detail: Action Panel</summary>

elixir-5
*Users can open the function documentation on the browser, paste the function call or copy it*
</details>

<details>
<summary>Search Modules: Text Search</summary>

elixir-6
*Users can search by name in the "Search Modules" command*
</details>

<details>
<summary>Search Modules: Module Functions</summary>

elixir-7
*After selecting a module, users can filter by type and search by name, and only the module's functions, types, macros and callbacks will be taken into consideration*
</details>

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/elixir/doc_builder.exs [hunk 1]
```diff
@@ -0,0 +1,255 @@
+# Script responsible for generating .ts files for each of Elixir's modules documentation.
```
Comment (dinocosta) [inline extensions/elixir/doc_builder.exs:1]: Not sure if this file should stay as part of the extension. It is used to generate the documentation files, for example, `src/docs/atom.ts`, but after those files are generated it is not needed for the extension to work. As such, I'm ok with moving it to a Github Gist or a different repository 👍

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋

Could you look into adding `isLoading` for your lists, it has some empty state flicker at the moment 🙂

**Comment by dinocosta** [comment]
> Hi 👋
> 
> Could you look into adding `isLoading` for your lists, it has some empty state flicker at the moment 🙂

@pernielsentikaer Thank you so much for taking the time to review this! 🙇 

Didn't manage to find the time to look into this today, will see if I can do it tomorrow and will tag you once it's done, thanks!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Is something missing in the code, there is an empty file called `components/ListItem.tsx`

**Comment by dinocosta** [comment]
> Is something missing in the code, there is an empty file called `components/ListItem.tsx`

@pernielsentikaer Weird, I probably staged it on accident when creating the other components, sorry for letting this one slip by. I've pushed this commit – https://github.com/raycast/extensions/pull/15580/commits/cc38c713265c2dc10fb2c10acf2e2dc5ffff16bd – to remove it, thank you for your patience  🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by dinocosta** [comment]
> Hi 👋
> 
> Looks good to me, approved 🔥

Thank you so much for your reviews @pernielsentikaer ! 🙇

**Pull request merged**